### PR TITLE
#3576 beam_stuck_guard_improvement

### PR DESCRIPTION
**Pull request by edwazhao**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3630)

Adding two parameters to gradle settings:
min_cpu_usage, the min cpu usage when checking the stuck guard shell
max_inactive_time_interval, the max time interval for no logs output when checking the stuck guard shell

### CODE REVIEW

File: aws/build.gradle [hunk 1]
```diff
@@ -112,6 +112,8 @@ task deploy(type: AWSLambdaInvokeTask) {
   "max_cores": ${ext.getParameterValue('maxCores') ?: 0},
   "min_memory": ${ext.getParameterValue('minMemory') ?: 0},
   "max_memory": ${ext.getParameterValue('maxMemory') ?: 0},
+  "min_cpu_usage": ${ext.getParameterValue('minCpuUsage') ?: 1},
+  "max_inactive_time_interval": ${ext.getParameterValue('maxInactiveTimeInterval') ?: 5},
```
Comment (nikolayilyin) [inline aws/build.gradle]: I would change the names of parameters to include 'stuckGuard', i.e. stuckGuardMinCpuUsage and stuckGuardMaxInactiveTimeInterval to reduce the possible confusion.

**Review by nikolayilyin — CHANGES_REQUESTED** [review]
Apart from a nitpick the changes also require to change the documentation, the deploy command is described in there - docs/developers.rst, let's add these two parameters into documentation

File: aws/build.gradle (hunk 1 above)
Comment (edwazhao) [reply aws/build.gradle]: Hi @nikolayilyin , I have changed the parameter names, please take a look again.

For the documentation, I couldn't find the correct place, do you mean here?
https://beam.readthedocs.io/en/latest/developers.html#automated-cloud-deployment with the parameter part?

Comment (nikolayilyin) [reply aws/build.gradle]: Yes, the link you shared contains the description of 'deploy' command, let's add the description of new parameters to that comand

Comment (edwazhao) [reply aws/build.gradle]: Hi @nikolayilyin , I did a commit to add description of the deploy command.
And also I did a rebase from develop branch, but now the changes of PR looks mess, only three files is changed by me
image

Do you have any suggestions of the default values? stuckGuardMinCpuUsage:1, stuckGuardMaxInactiveTimeInterval:5

Comment (nikolayilyin) [reply aws/build.gradle]: The default values looks OK for me, let's keep them as 5 and 1 for now

Comment (edwazhao) [reply aws/build.gradle]: Can we close the issue and merge the PR? Or what is the process for BEAM project?

**Pull request closed without merging**